graphene_sqlalchemy/tests/test_batching.py

```diff
@@ -226,3 +256,257 @@ def test_one_to_one(session_factory):
         },
       ],
     }
+
+
+def test_one_to_many(session_factory):
+    session = session_factory()
+
+    reporter_1 = Reporter(
+      first_name='Reporter_1',
+    )
+    session.add(reporter_1)
+    reporter_2 = Reporter(
+      first_name='Reporter_2',
+    )
+    session.add(reporter_2)
+
+    article_1 = Article(headline='Article_1')
+    article_1.reporter = reporter_1
+    session.add(article_1)
+
+    article_2 = Article(headline='Article_2')
+    article_2.reporter = reporter_1
+    session.add(article_2)
+
+    article_3 = Article(headline='Article_3')
+    article_3.reporter = reporter_2
+    session.add(article_3)
+
+    article_4 = Article(headline='Article_4')
+    article_4.reporter = reporter_2
+    session.add(article_4)
+
+    session.commit()
+    session.close()
+
+    schema = get_schema()
+
+    with mock_sqlalchemy_logging_handler() as sqlalchemy_logging_handler:
+        # Starts new session to fully reset the engine / connection logging level
+        session = session_factory()
+        result = schema.execute("""
+          query {
+            reporters {
+              firstName
+              articles(first: 2) {
+                edges {
+                  node {
+                    headline
+                  }
+                }
+              }
+            }
+          }
+        """, context_value={"session": session})
+        messages = sqlalchemy_logging_handler.messages
+
+    assert len(messages) == 5
+
+    if is_sqlalchemy_version_less_than('1.3'):
```

Comment:
Should we not also test the SQL generated in 1.2 ?

Reply:
Generally speaking, I think we should minimize this type of assertions because they are going to be a little annoying to maintain down the road. If anyone updates the base models (e.g. add another column), these tests will have to be updated as well because the generated SQL will be different. 

For `1.3`, it makes sense to explicitly test the SQL because it makes it clear that 5 SQL messages in this context corresponds to 2 `SELECT` statements, that is batching is enabled. That's why it's enough for `1.2` to only assert there are 5 messages because we know it means that batching is enabled. There would be 7 messages (ie 3 SQL statements) without batching.

Reply:
To workaround the problem mentioned in my previous comment, I added some SQL assertions that should be easy to maintain.